### PR TITLE
Migrate `site-editor-inserter` e2e test to Playwright

### PR DESCRIPTION
**Pull request by alvitazwar**
## What?
Based on https://github.com/WordPress/gutenberg/pull/38570, part of https://github.com/WordPress/gutenberg/issues/38851. Migrate [site-editor-inserter.test.js](https://github.com/WordPress/gutenberg/blob/trunk/packages/e2e-tests/specs/site-editor/site-editor-inserter.test.js) to its Playwright counterpart.

## Why?
See https://github.com/WordPress/gutenberg/pull/38570 for its background and rationale.

This is split into a new PR for easier review.

##How?
See https://github.com/WordPress/gutenberg/pull/38570 for the proposed migration steps.

##Testing Instructions
npm run test:e2e:playwright /test/e2e/specs/site-editor/site-editor-inserter.spec.js

## Screencast

https://user-images.githubusercontent.com/55917380/193241408-10341ee2-a3df-4edd-ac7f-8d0b36dd7378.mov

### CODE REVIEW

**Comment by alvitazwar** [comment]
@kevin940726  can you please review this PR?

**Comment by alvitazwar** [comment]
@Mamaduka  Can you please help me to review this PR?

File: test/e2e/specs/site-editor/site-editor-inserter.spec.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/**
+ * WordPress dependencies
+ */
+const { test } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Block list view', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'emptytheme' );
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/site-editor-inserter.spec.js]: We should also delete all templates as seen in the original test:

```suggestion
		await Promise.all( [
			requestUtils.activateTheme( 'emptytheme' ),
			requestUtils.deleteAllTemplates( 'wp_template' ),
			requestUtils.deleteAllTemplates( 'wp_template_part' ),
		] );
```

File: test/e2e/specs/site-editor/site-editor-inserter.spec.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	test.beforeEach( async ( { admin } ) => {
+		await admin.visitSiteEditor();
+	} );
+
+	test( 'inserter toggle button should toggle global inserter', async ( {
+		page,
+	} ) => {
+		await page
+			.locator( 'role=button[name="Toggle block inserter"i]' )
+			.click();
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/site-editor-inserter.spec.js]: Nit:

```suggestion
		await page.click( 'role=button[name="Toggle block inserter"i]' );
```

File: test/e2e/specs/site-editor/site-editor-inserter.spec.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+		await page
+			.locator( '.edit-site-editor__inserter-panel' )
+			.waitFor( { state: 'visible' } );
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/site-editor-inserter.spec.js]: Using class selectors is discouraged. Consider using role selectors instead:

```suggestion
		await expect(
			page.locator( 'role=region[name="Block Library"i]' )
		).toBeVisible();
```

File: test/e2e/specs/site-editor/site-editor-inserter.spec.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+		await page
+			.locator( 'role=button[name="Toggle block inserter"i]' )
+			.click();
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/site-editor-inserter.spec.js]: Nit: 
```suggestion
		await page.click( 'role=button[name="Toggle block inserter"i]' );
```

File: test/e2e/specs/site-editor/site-editor-inserter.spec.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+		await page
+			.locator( '.edit-site-editor__inserter-panel' )
+			.waitFor( { state: 'hidden' } );
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/site-editor-inserter.spec.js]: Same here:

```suggestion
		await expect(
			page.locator( 'role=region[name="Block Library"i]' )
		).toBeHidden();
```

File: test/e2e/specs/site-editor/site-editor-inserter.spec.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
+/**
+ * WordPress dependencies
+ */
+const { test } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Block list view', () => {
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/site-editor-inserter.spec.js]: Copy-paste error?

```suggestion
test.describe( 'Site Editor Inserter', () => {
```

File: test/e2e/specs/site-editor/site-editor-inserter.spec.js (hunk 1 above)
Comment (alvitazwar) [reply test/e2e/specs/site-editor/site-editor-inserter.spec.js]: Addressed as suggested

File: test/e2e/specs/site-editor/site-editor-inserter.spec.js (hunk 3 above)
Comment (alvitazwar) [reply test/e2e/specs/site-editor/site-editor-inserter.spec.js]: Implemented

File: test/e2e/specs/site-editor/site-editor-inserter.spec.js (hunk 6 above)
Comment (alvitazwar) [reply test/e2e/specs/site-editor/site-editor-inserter.spec.js]: Yes, it was a copy-paste mistake I was nervous. thanks

File: test/e2e/specs/site-editor/site-editor-inserter.spec.js (hunk 2 above)
Comment (alvitazwar) [reply test/e2e/specs/site-editor/site-editor-inserter.spec.js]: Implemented

**Comment by alvitazwar** [comment]
@kevin940726  please check now

File: test/e2e/specs/site-editor/site-editor-inserter.spec.js (hunk 3 above)
Comment (kevin940726) [reply test/e2e/specs/site-editor/site-editor-inserter.spec.js]: I don't think this is right? I think some code are accidentally removed?

Comment (alvitazwar) [reply test/e2e/specs/site-editor/site-editor-inserter.spec.js]: Sorry for the mistake its fixed now

**Comment by alvitazwar** [comment]
Addressed locator feedback

File: test/e2e/specs/site-editor/site-editor-inserter.spec.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+		await admin.visitSiteEditor();
+	} );
+
+	test( 'inserter toggle button should toggle global inserter', async ( {
+		page,
+	} ) => {
+		await page.click( 'role=button[name="Toggle block inserter"i]' );
+		await expect(
+			page.locator( 'role=region[name="Block Library"i]' )
+		).toBeVisible();
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/site-editor-inserter.spec.js:33]: We still need the other steps: clicking on the button again and expecting that it's hidden.

Comment (alvitazwar) [reply test/e2e/specs/site-editor/site-editor-inserter.spec.js:33]: I added both the steps and added role locators for all the assertions. Thanks for guiding me and helping me out

File: test/e2e/specs/site-editor/site-editor-inserter.spec.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+		await page.click( 'role=button[name="Toggle block inserter"i]' );
+
+		// Visibility check
+		await expect(
+			page.locator(
+				'role=searchbox[name="Search for blocks and patterns"i]'
+			)
+		).toBeVisible();
+		await page.click( 'role=button[name="Toggle block inserter"i]' );
+		//Hidden State check
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/site-editor-inserter.spec.js:35]:
```suggestion
		// Hidden state check.
```

File: test/e2e/specs/site-editor/site-editor-inserter.spec.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+	test.beforeEach( async ( { admin } ) => {
+		await admin.visitSiteEditor();
+	} );
+
+	test( 'inserter toggle button should toggle global inserter', async ( {
+		page,
+	} ) => {
+		await page.click( 'role=button[name="Toggle block inserter"i]' );
+
+		// Visibility check
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/site-editor-inserter.spec.js:28]:
```suggestion
		// Visibility check.
```

**Review by kevin940726 — APPROVED** [review]
LGTM 👍 Thanks!

**Comment by alvitazwar** [comment]
Thanks @kevin940726  for the support

**Comment by kevin940726** [comment]
There seems to have conflicts, do you mind rebasing? :)

**Comment by alvitazwar** [comment]
@kevin940726  Please check now

**Review by kevin940726 — COMMENTED** [review]
Be sure to remove the original test and then we should be all set! Thanks!

**Review by kevin940726 — APPROVED** [review]
Nice one, thanks!

**Comment by talldan** [comment]
The PHP test failures are unrelated (there's a fix in https://github.com/WordPress/gutenberg/pull/45265), so I'll move ahead with merging this one.

**Pull request merged**